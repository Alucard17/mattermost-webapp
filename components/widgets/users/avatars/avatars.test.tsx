// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {shallow} from 'enzyme';

import Avatar from '../avatar';

import Avatars from './avatars';

const users = [
    {
        url: 'test-url-1',
        username: 'jesus.espino',
        name: 'Jesus Espino',
    },
    {
        url: 'test-url-2',
        username: 'johnny.depp',
        name: 'Johnny Depp',
    },
    {
        url: 'test-url-3',
        username: 'bilbo.baggins',
        name: 'Bilbo Baggins',
    },
    {
        url: 'test-url-4',
        username: 'michael.hall',
        name: 'Anthony Michael Hall',
    },
    {
        url: 'test-url-5',
        username: 'kathy.baker',
        name: 'Kathy Baker',
    },
];

describe('components/widgets/users/Avatars', () => {
    test('should match the snapshot', () => {
        const wrapper = shallow(
            <Avatars
                size='xl'
                users={users}
            />,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should properly count overflow', () => {
        const wrapper = shallow(
            <Avatars
                size='xl'
                users={users}
            />,
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(Avatar).find({url: 'test-url-1'}).exists()).toBe(true);
        expect(wrapper.find(Avatar).find({url: 'test-url-2'}).exists()).toBe(true);
        expect(wrapper.find(Avatar).find({url: 'test-url-3'}).exists()).toBe(true);
        expect(wrapper.find(Avatar).find({url: 'test-url-4'}).exists()).toBe(false);
        expect(wrapper.find(Avatar).find({url: 'test-url-5'}).exists()).toBe(false);
    });
});
