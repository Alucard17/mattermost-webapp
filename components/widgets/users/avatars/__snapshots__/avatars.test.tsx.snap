// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/widgets/users/Avatars should match the snapshot 1`] = `
<div
  className="Avatars Avatars___xl"
  onMouseLeave={[Function]}
>
  <Memo(SimpleTooltip)
    animation={true}
    content="Jesus Espino"
    id="name-jesus.espino"
    key="test-url-1"
    onEntered={[Function]}
  >
    <Memo(Avatar)
      size="xl"
      tabIndex={0}
      url="test-url-1"
      username="jesus.espino"
    />
  </Memo(SimpleTooltip)>
  <Memo(SimpleTooltip)
    animation={true}
    content="Johnny Depp"
    id="name-johnny.depp"
    key="test-url-2"
    onEntered={[Function]}
  >
    <Memo(Avatar)
      size="xl"
      tabIndex={0}
      url="test-url-2"
      username="johnny.depp"
    />
  </Memo(SimpleTooltip)>
  <Memo(SimpleTooltip)
    animation={true}
    content="Bilbo Baggins"
    id="name-bilbo.baggins"
    key="test-url-3"
    onEntered={[Function]}
  >
    <Memo(Avatar)
      size="xl"
      tabIndex={0}
      url="test-url-3"
      username="bilbo.baggins"
    />
  </Memo(SimpleTooltip)>
  <Memo(SimpleTooltip)
    animation={true}
    content="Anthony Michael Hall, Kathy Baker"
    id="names-overflow"
    onEntered={[Function]}
  >
    <Memo(Avatar)
      size="xl"
      tabIndex={0}
      text="+2"
    />
  </Memo(SimpleTooltip)>
</div>
`;

exports[`components/widgets/users/Avatars should properly count overflow 1`] = `
<div
  className="Avatars Avatars___xl"
  onMouseLeave={[Function]}
>
  <Memo(SimpleTooltip)
    animation={true}
    content="Jesus Espino"
    id="name-jesus.espino"
    key="test-url-1"
    onEntered={[Function]}
  >
    <Memo(Avatar)
      size="xl"
      tabIndex={0}
      url="test-url-1"
      username="jesus.espino"
    />
  </Memo(SimpleTooltip)>
  <Memo(SimpleTooltip)
    animation={true}
    content="Johnny Depp"
    id="name-johnny.depp"
    key="test-url-2"
    onEntered={[Function]}
  >
    <Memo(Avatar)
      size="xl"
      tabIndex={0}
      url="test-url-2"
      username="johnny.depp"
    />
  </Memo(SimpleTooltip)>
  <Memo(SimpleTooltip)
    animation={true}
    content="Bilbo Baggins"
    id="name-bilbo.baggins"
    key="test-url-3"
    onEntered={[Function]}
  >
    <Memo(Avatar)
      size="xl"
      tabIndex={0}
      url="test-url-3"
      username="bilbo.baggins"
    />
  </Memo(SimpleTooltip)>
  <Memo(SimpleTooltip)
    animation={true}
    content="Anthony Michael Hall, Kathy Baker"
    id="names-overflow"
    onEntered={[Function]}
  >
    <Memo(Avatar)
      size="xl"
      tabIndex={0}
      text="+2"
    />
  </Memo(SimpleTooltip)>
</div>
`;
